raster-foundry/raster-foundry
# Allow geojson annotation import export

## Overview

This PR enables geojson annotations to be imported from and exported to local drives. The upload supports multiple geojson files. Uploaded annotations can be edited, deleted, or filtered just as the same as other new annotations. New annotations can be added as well.  It also enables all or filtered annotations to be exported/downloaded to local drives in geojson format.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~
~- [ ] PR has a name that won't get you publicly shamed for vagueness~

### Demo

* Geojson file(s) upload
<img width="1680" alt="screen shot 2017-08-07 at 5 28 54 pm" src="https://user-images.githubusercontent.com/16109558/29046648-61bf18d0-7b96-11e7-821d-105d1121fa8d.png">

* Import after upload
<img width="1680" alt="screen shot 2017-08-07 at 5 29 00 pm" src="https://user-images.githubusercontent.com/16109558/29046665-793427bc-7b96-11e7-99e8-e3767e84b7e4.png">
<img width="1680" alt="screen shot 2017-08-07 at 5 29 06 pm" src="https://user-images.githubusercontent.com/16109558/29046668-7cb0a5aa-7b96-11e7-9b80-b214b242a0b8.png">

* Add/edit/delete/filter won't be affected
<img width="1680" alt="screen shot 2017-08-07 at 5 31 00 pm" src="https://user-images.githubusercontent.com/16109558/29046712-ab2cfbc2-7b96-11e7-914d-1da8488b79d9.png">

* Exported geojson with newly added "Delaware"
<img width="1680" alt="screen shot 2017-08-07 at 5 31 15 pm" src="https://user-images.githubusercontent.com/16109558/29046731-be098e54-7b96-11e7-9bd3-55ea7afc2ff3.png">



### Notes

* The second commit of this PR is for import and export.  I will rebase once the previous PR containing the first commit is merged.
* "Upload" and "Import" are separated in two actions, because there should be some actions (like select properties to match our "label", "id", "description" fielts, etc.) in between - which will be done in the next round for polishing.


## Testing Instructions

 * Go to `projects/edit/:project_id/annotate`
 * Import, add, edit, delete, filter, export. See if it behaves as you expect.

Closes #2281, #2282
